r/tests/testthat/test-dplyr-funcs-type.R

```diff
@@ -875,6 +877,134 @@ test_that("as.Date() converts successfully from date, timestamp, integer, char a
   )
 })
 
+test_that("as_date()", {
+  test_df <- tibble::tibble(
+    posixct_var = as.POSIXct("2022-02-25 00:00:01", tz = "Pacific/Marquesas"),
+    date_var = as.Date("2022-02-25"),
+    difference_date = ymd_hms("2010-08-03 00:50:50", tz = "Pacific/Marquesas"),
+    character_ymd_var = "2022-02-25 00:00:01",
+    character_ydm_var = "2022/25/02 00:00:01",
+    integer_var = 32L,
+    integerish_var = 32,
+    double_var = 34.56
+  )
+
```

Comment:
[Jon's comment](https://github.com/apache/arrow/pull/12677#discussion_r836786450) on the original PR:
> This fixture looks very similar (though subtly different in some ways) as the one for as.Date() would it make sense to combine them into a single test_that block so they can use one fixture and we don't have to redefine it + make sure we update it in both places, etc?

>If that's not feasible, should we move the fixture out to helpers-data.R so that it's reusable?

Reply:
Done. I combined the tests for `as.Date()` and `as_date()` into a single `test_that` block.